### ISSUE TITLE
Allow passing command line parameters to vstest and mstest

### ISSUE THREAD

**Issue by Steinblock** [axodox/AxoTools#14]
Eventually a gui for configuring vstest and mstest would be nice.
However in the meantime I would suggest the following

a) Allow passing additional parameters to the test runner as a string in `Settings -> Test Runners`

This is what I currently use for opencover and I would love to have these settings in AxoTools, too.

```
-targetargs:"/Parallel /InIsolation /TestCaseFilter:\"TestCategory!=excel\"" ...
```

b) Please output the whole command line in output window

Currently it just prints 

```
Coverage executor has started.
Executing: C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe
```

For debugging it would be nice to have a "copy&paste" command line that I can execute in a console window.

**Comment by axodox** [comment]
The next version of AxoCover will no longer use the VSTest/MSTest runners, but it will use its own runner instead, which will still call the same test adapters and it will offer the main options. This change is made, because the interoperation with MS and especially VSTest was problematic, this will also mean that the tests will run with the same runner in Visual Studio 2012 - 2017 which will produce more consistent results. Debugging will also use this runner, and no longer revert to the VisualStudio one - again for consistency. The command line arguments will no longer be for much use, because AxoCover's runner is using the network to configure and execute test runs.

As Microsoft's new open source test-platform progresses, eventually AxoCover might use that. But for now it will use its own, lightweight solution. 

You may try it [here](https://github.com/axodox/AxoTools/releases): xNUnitSupport.
Additional options will be added though, as it was requested in an other issue too.

**Comment by axodox** [comment]
I will get back to this, once the 1.1 version is released.

**Comment by axodox** [comment]
Debugging is now also built in, so no longer need to copy-paste command line. Also mstest and vstest are no longer used. I will close this.

**Comment by Steinblock** [comment]
The main purpose of this issue was to improve AxoCover performance with `/Parallel`

If this issue is closed, Is there an option in AxoCover to run tests in Parallel?

Currently my test run takes
* about 30 secs with Visual Studio (checked `Run tests in parallel`), 
* about 65 secs with `OpenCover.Console.exe -target:"vstest.console.exe" -targetargs:"/Parallel /Settings:default.runsettings Project1.Tests.dll Project2.Tests.dll Project3.Tests.dll Project4.Tests.dll"` 
* about 1 min 50 secs with AxoCover `Run`
* and about 5 min 20 secs with AxoCover `Cover` 

so with parallel test execution I could improve the performance by 500% wich is perfectly supported by opencover.

Note: `vstest /Parallel` does not run tests inside a single project in parallel but executes echt test assembly in parallel. My Solution has 9 test projects so the test run duration is determined by the largest test project.

**Comment by axodox** [comment]
Currently AxoCover's runner does not have parallel execution implemented for separate assemblies. I could look into trying how it would work if I would change the foreach on assemblies to Parallel.Foreach in AxoCover.Runner.

As for single threaded performance currently AxoCover ships with the Cover By Test feature enabled by default, which has an effect on performance. You may change it in the settings to make test runs faster. Note: this will disable some features.
